app/src/main/java/org/mozilla/focus/home/HomeScreen.kt

```diff
@@ -11,32 +11,25 @@ import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.unit.dp
-import mozilla.components.feature.top.sites.TopSite
-import org.mozilla.focus.topsites.TopSiteMenuItem
+import kotlinx.coroutines.DelicateCoroutinesApi
+import mozilla.components.lib.state.ext.observeAsComposableState
+import org.mozilla.focus.components
 import org.mozilla.focus.topsites.TopSites
 
 /**
  * The home screen.
- *
- * @param topSites List of [TopSite] to display.
- * @param topSitesMenuItems List of [TopSiteMenuItem] to display in a top site dropdown menu.
- * @param onTopSiteClick Invoked when the user clicks on a top site.
  */
+@OptIn(DelicateCoroutinesApi::class)
 @Composable
-fun HomeScreen(
-    topSites: List<TopSite>,
-    topSitesMenuItems: List<TopSiteMenuItem>,
-    onTopSiteClick: (TopSite) -> Unit = {}
-) {
+fun HomeScreen() {
+    val topSitesState = components.appStore.observeAsComposableState { state -> state.topSites }
+    val topSites = topSitesState.value!!
```

Comment:
If you don't need this value here then I would avoid calling `.value` because afaik this will mean this scope is subscribed to it and will get recomposed whenever this value changes. But if you only need it down the tree then read the value only down the tree: Then only this subtree will need to get recomposed.